repose-aggregator/components/filters/keystone-v2/src/main/scala/org/openrepose/filters/keystonev2/KeystoneV2Filter.scala

```diff
@@ -409,6 +409,8 @@ class KeystoneV2Filter @Inject()(configurationService: ConfigurationService,
                   case Success(newAdminToken) => requestHandler.getGroups(newAdminToken, validToken.userId)
                   case Failure(x) => Failure(IdentityAdminTokenException("Unable to reacquire admin token", x))
                 }
+              case _: InvalidTokenException if Option(keystoneV2Config.getIdentityService.getSetRackersGroups).isDefined && validToken.roles.exists(_.equalsIgnoreCase("racker")) =>
```

Comment:
I made this a case-insensitive check, which seems reasonable, but worth pointing out. Actually, I could see it either way.

To be clear, roles are case sensitive, and "Racker" seems to be the role for Rackers in Rackspace Identity.